app/components/blacklight/document_component.rb

```diff
@@ -90,7 +90,7 @@ def initialize(document: nil, partials: nil,
                    id: nil, classes: [], component: :article, title_component: nil,
                    counter: nil, document_counter: nil, counter_offset: 0,
                    show: false, **args)
-      @presenter = document || args[self.class.collection_parameter]
+      @presenter = document || args[self.class.__vc_collection_parameter]
```

Comment:
In what cases do we need to fall back to `self.class.__vc_collection_parameter`?  It would be nice if we could avoid relying on a method that is supposed to be more internal to ViewComponent.

Or maybe we could do this in a way that supports both 3.x and 4.x? (e.g. add a getter method to `Blacklight::Component` that wraps `collection_parameter` when the gem version is 3.x and `__vc_collection_parameter` when it is 4.x.)  That could be nice, so implementers only have to do one major gem version upgrade at a time.

Reply:
Ah, and I just saw #3714 which resolves both of my questions as far as `DocumentComponent` goes. :-)